mlir/lib/Target/LLVM/ModuleToObject.cpp

```diff
@@ -39,32 +39,32 @@ ModuleToObject::ModuleToObject(Operation &module, StringRef triple,
     : module(module), triple(triple), chip(chip), features(features),
       optLevel(optLevel) {}
 
+ModuleToObject::~ModuleToObject() = default;
+
 Operation &ModuleToObject::getOperation() { return module; }
 
-std::unique_ptr<llvm::TargetMachine> ModuleToObject::createTargetMachine() {
+std::optional<llvm::TargetMachine *>
+ModuleToObject::getOrCreateTargetMachine() {
   std::string error;
```

Comment:
Is there a reason why the cached `targetMachine` is not immediately returned? 
```suggestion
  if (targetMachine)
    return targetMachine.get();
  std::string error;
```

Reply:
Thanks!